flytekit/tools/repo.py

```diff
@@ -91,6 +91,10 @@ def package(
 
         # If Fast serialization is enabled, then an archive is also created and packaged
         if fast:
+            # If output exists and is a path within source, delete it so as to not re-bundle it again.
```

Comment:
`output_tmpdir` is just a vessel for contents of `source` to be packaged into. Even if we save it somewhere else, it doesn’t matter since the size of `.tar.gz` will still double as the real underlying issue is `output` being a subpath of `source`. I will make it so that if `output` is not supplied, then save it to a temporary directory. Else, use it. With that said, I still think we need a check for `output` being within `source` — and maybe display an error there — so as to prevent it from saving altogether — This is for the case when the user supplies `output` such that it is still inside `source` , so it is worth handling I guess.